provider/provider.go

```diff
@@ -60,6 +60,14 @@ func (p *Provider) Init(_ string, dsn string, _ bool) error {
 	// Create tables
 	m := NewMigrator(p.db, p.Logger)
 	for _, t := range p.ResourceMap {
+
+		// validate table
+		validationErr := m.ValidateTable(t)
```

Comment:
maybe its unnecessary to have the table validator part of the migrator, it does have a different purpose. We can have a defaultTableValidator and use that for now, or even just a function, and move that into a struct in the future.